### PR TITLE
Borda pr

### PR DESCRIPTION
**Pull request by bmacias01**
Andrew and I added Borda Election code in the election_types.py file.

### CODE REVIEW

File: src/votekit/election_types.py [hunk 1]
```diff
@@ -78,6 +78,78 @@ def run_step(self, profile: PreferenceProfile) -> tuple[PreferenceProfile, Outco
...
+        #     curr_round=1,
+        #     elected=winners,
+        #     eliminated=sorted_borda[seats:],
+        #     remaining=[],
+        #     profile=profile,
+        #     winner_votes=winner_votes,
+        #     previous=None
+        # )
+
+    def run_borda_election(self):
```
Comment (jgibson517) [inline src/votekit/election_types.py:149]: lets unpack this before we return for clarity

i.e 
```suggestion
    def run_borda_election(self):
        profile, outcome = self.run_borda_step()
        return outcome 
     
    
```

**Review by jgibson517 — CHANGES_REQUESTED** [review]

File: src/votekit/election_types.py [hunk 2]
```diff
@@ -78,6 +78,78 @@ def run_step(self, profile: PreferenceProfile) -> tuple[PreferenceProfile, Outco
...
+        candidates_ballots = {}  # {candidate : [ballots mentioning candidate]}
+
+        for ballot in self.profile.get_ballots():
+            frequency = ballot.weight
+            index = 0
+            for candidate in ballot.ranking:
+                candidate = str(candidate)
+
+                if candidate not in candidate_rank_freq:
+                    candidate_rank_freq[candidate] = [
```
[the remark this answers is not in the document]
Comment (andrewmhong) [reply src/votekit/election_types.py:105]: Done

File: src/votekit/election_types.py [hunk 3]
```diff
@@ -78,6 +78,78 @@ def run_step(self, profile: PreferenceProfile) -> tuple[PreferenceProfile, Outco
...
+
+        borda_scores = {}  # {candidate : int borda_score}
+        candidate_rank_freq = (
+            {}
+        )  # {candidate : [1st rank total, 2nd rank total,..., n rank total]}
+        candidates_ballots = {}  # {candidate : [ballots mentioning candidate]}
+
+        for ballot in self.profile.get_ballots():
+            frequency = ballot.weight
+            index = 0
```
[the remark this answers is not in the document]
Comment (andrewmhong) [reply src/votekit/election_types.py:100]: Done

File: src/votekit/election_types.py [hunk 4]
```diff
@@ -78,6 +78,78 @@ def run_step(self, profile: PreferenceProfile) -> tuple[PreferenceProfile, Outco
...
+            for candidate in ballot.ranking:
+                candidate = str(candidate)
+
+                if candidate not in candidate_rank_freq:
+                    candidate_rank_freq[candidate] = [
+                        0 for _ in range(len(ballot.ranking))
+                    ]
+                    candidate_rank_freq[candidate][index] = frequency
+                else:
+                    candidate_rank_freq[candidate][index] += frequency
```
[the remark this answers is not in the document]
Comment (andrewmhong) [reply src/votekit/election_types.py:110]: Done

File: src/votekit/election_types.py [hunk 5]
```diff
@@ -78,6 +78,78 @@ def run_step(self, profile: PreferenceProfile) -> tuple[PreferenceProfile, Outco
...
+                if candidate not in candidates_ballots:
+                    candidates_ballots[candidate] = []
+                    candidates_ballots[candidate].append(ballot)
+                else:
+                    candidates_ballots[candidate].append(ballot)
+                index += 1
+
+        for key in candidate_rank_freq:
```
[the remark this answers is not in the document]
Comment (andrewmhong) [reply src/votekit/election_types.py:118]: Done

File: src/votekit/election_types.py [hunk 6]
```diff
@@ -78,6 +78,78 @@ def run_step(self, profile: PreferenceProfile) -> tuple[PreferenceProfile, Outco
...
+
+        winners = sorted_borda[: self.seats]
+
+        # get winner_votes
+        # TO-DO: Adjust Outcome class to new args
+        winner_votes = {}
+        for winner in winners:
+            winner_votes[winner] = candidates_ballots[winner]
+
+        return PreferenceProfile(ballots=self.profile.get_ballots()), Outcome(
```
[the remark this answers is not in the document]
Comment (andrewmhong) [reply src/votekit/election_types.py:133]: Done

**Pull request closed without merging**